src/Apps/MyCollection/Routes/EditArtwork/Utils/createOrUpdateArtwork.ts

```diff
@@ -0,0 +1,47 @@
+import { MyCollectionCreateArtworkInput } from "__generated__/useCreateArtworkMutation.graphql"
+import { MyCollectionUpdateArtworkInput } from "__generated__/useUpdateArtworkMutation.graphql"
+import { useCreateArtwork } from "../Mutations/useCreateArtwork"
+import { useUpdateArtwork } from "../Mutations/useUpdateArtwork"
+
+export type ArtworkInput =
+  | MyCollectionCreateArtworkInput
+  | MyCollectionUpdateArtworkInput
+
+export const useCreateOrUpdateArtwork = () => {
+  const { submitMutation: createArtwork } = useCreateArtwork()
+  const { submitMutation: updateArtwork } = useUpdateArtwork()
+
+  const createOrUpdateArtwork = async (artwork: ArtworkInput) => {
+    console.log("createOrUpdateArtwork", artwork)
+
+    let artworkId: string | undefined
+
+    if ((artwork as MyCollectionUpdateArtworkInput).artworkId) {
```

Comment:
**thought:** I am wondering if there is a way for us to avoid this casting since I find it dangerous especially in this case

Reply:
You're right, there is a better way! If we check for `"artworkId" in artwork` instead, TypeScript will type `artwork` correctly as `MyCollectionUpdateArtworkInput` inside of the if block 👏